students/DiannaTingg/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,185 @@
+
+
+def _import_csv(filename):
+    """
+    Returns a list of dictionaries.  One dictionary for each row of data in a csv file.
+    :param filename: csv file
+    :return: list of dictionaries
+    """
+
+    with open(filename, newline="") as csvfile:
+        dict_list = []
+
+        csv_data = csv.reader(csvfile)
+
+        headers = next(csv_data, None)  # Save the first line as the headers
+
+        if headers[0].startswith("ï»¿"):  # Check for weird formatting
+            headers[0] = headers[0][3:]
+
+        for row in csv_data:
+            row_dict = {}
+
+            for index, column in enumerate(headers):
+                row_dict[column] = row[index]
+
+            dict_list.append(row_dict)
+
+        return dict_list
+
+
+def _add_bulk_data(collection, directory_name, filename):
+    file_path = os.path.join(directory_name, filename)
+
+    try:
+        collection.insert_many(_import_csv(file_path), ordered=False)
+        return 0
+
+    except pymongo.errors.BulkWriteError as bwe:
+        print(bwe.details)
+        return len(bwe.details["writeErrors"])
+
+
+def import_data(db, directory_name, products_file, customers_file, rentals_file):
+    """
+    Takes a directory name and three csv files as input.  Creates and populates a new MongoDB.
+    :param db: MongoDB
+    :param directory_name: directory name for files.  Use "" for current directory.
+    :param products_file: csv file with product data
+    :param customers_file: csv file with customer data
+    :param rentals_file: csv file with rentals data
+    :return: Tuple with record count for products, customers, rentals added (in that order) and
+             tuple with count of errors that occurred for products, customers, rentals (in that order).
+    """
+
+    products = db["products"]
+    products_errors = _add_bulk_data(products, directory_name, products_file)
+
+    customers = db["customers"]
+    customers_errors = _add_bulk_data(customers, directory_name, customers_file)
+
+    rentals = db["rentals"]
+    rentals_errors = _add_bulk_data(rentals, directory_name, rentals_file)
+
+    record_count = (db.products.count_documents({}), db.customers.count_documents({}), db.rentals.count_documents({}))
+
+    error_count = (products_errors, customers_errors, rentals_errors)
+
+    return record_count, error_count
+
+
+def show_available_products(db):
+    """
+    Returns a dictionary for each product listed as available.
+    :param db: MongoDB
+    :return: Dictionary with product_id, description, product_type, quantity_available.
+    """
+
+    available_products = {}
+
+    for product in db.products.find():
+        if int(product["quantity_available"]) > 0:
+
+            product_dict = {"description": product["description"],
+                            "product_type": product["product_type"],
+                            "quantity_available": product["quantity_available"]}
+
+            available_products[product["product_id"]] = product_dict
```

Comment:
product and product_dict look very similar. Are the keys/values in product that you don't want in product_dict? Is that why you aren't just making a copy? If it is just one key, you could make a copy and delete that key. Or do a comprehension (where I'm putting it directly into the available_products dictionary)
```
available_products[product["product_id"]] = {key: value for key, value in product.items() if key is not "quantity_available"}
```
It is a pretty long line, so may be better to save it first, then enter into dict. 
This can be done in show_rentals, as well, I think.